src/planner/binder/query_node/bind_recursive_cte_node.cpp

```diff
@@ -36,6 +36,18 @@ unique_ptr<BoundQueryNode> Binder::BindNode(RecursiveCTENode &statement) {
 
 	result->right_binder = Binder::CreateBinder(context, this);
 
+	if (!statement.key_targets.empty()) {
+		result->recurring_index = GenerateTableIndex();
+		// Retrieves the recursive CTE information in order to add it to the own context for the recurring table.
+		vector<std::reference_wrapper<CommonTableExpressionInfo>> current_cte_info = FindCTE(statement.ctename);
+		string recurring_name = "recurring_" + statement.ctename;
```

Comment:
Could we add a test when this name already exists as another CTE? e.g. something like:


```sql
WITH RECURSIVE CTE recurring_tbl AS (SELECT 42),
   tbl(a,b) USING KEY (a) AS (...)
SELECT * FROM tbl;
```

Not sure what the behavior here should be, error perhaps?

Reply:
We had another discussion about this problem in our chair today, and we had three possible solutions.
 1. We will throw an error that we have ambiguous names
 2. We allow it and have "variable shadowing". We will only be able to refer to the recurring table, not the cte.
 3. We use a different syntax to refer to the recurring table, which is the most popular version. `RECURRING tbl` would replace `recurring_tbl` and would be used like `ONLY`. This means that a correct reference would look like the following:

```
WITH RECURSIVE tbl USING KEY a AS (
...
UNION
SELECT ...
FROM tbl, RECURRING tbl AS [name]
```

What do you think about these options?

Reply:
I did implement option 3.